imports/ui/pages/admin/admin_dashboard.jsx

```diff
@@ -114,99 +136,131 @@ class _AdminDashboard extends Component {
+                </div>
+                <input className='form-control' id='emailField' type='email' data-name='email' onChange={this.setValue} placeholder='Email' />
+                <br />
+                <input className='form-control' id='passwordField' type='password' data-name='password' onChange={this.setValue} placeholder='Password' /><br />
+                <div><input className='form-control' type='password' data-name='password_verify' onChange={this.setValue} placeholder='Retype Password' /><br /></div>
+                <div className='spacer1'>&nbsp;</div>
+                <input type='submit' id='submitButton' className='btn btn-primary btn-block' value='Submit' />
+              </div>
+            </form>
           </div>
-        </form>
-        { this.state.profileViewModal
-          ? <ProfileViewModal
-            user={this.state.userToView}
-            done={this.toggleProfileViewModal} />
-        : '' }
+        </div>
+
+        <div>
+          <h1>Users (with elevated permissions first)</h1>
+          <table className='table table-bordered'>
+            <tbody>
+              <tr>
+                <th>Name</th>
+                <th>Email</th>
+                <th>Courses</th>
+                <th>Change Role</th>
+              </tr>
+
+              {
+                this.props.allUsers.map((u) => {
+                  let courseList = ''
+                  if (u.profile.courses && this.props) {
+                    u.profile.courses.forEach(function (cId) {
+                      courseList += this.props.courseNames[cId] ? this.props.courseNames[cId] + ' ' : ''
+                    }.bind(this))
+                  }
+                  return (<tr key={u._id}>
+                    <td>
+                      <a href='#' onClick={(e) => this.toggleProfileViewModal(u)}>{u.getName()}</a>
+                    </td>
+                    <td>{u.getEmail()} &nbsp;&nbsp; {u.emails[0].verified ? '(verified)'
+                        : <a href='#' onClick={(e) => this.verifyUserEmail(u.getEmail())}>Verify</a>}
+                    </td>
+                    <td>{courseList}</td>
+                    <td>
+                      <select onChange={(e) => this.saveRoleChange(u._id, e.target.value)} value={u.getRole()}>
+                        { Object.keys(ROLES).map((r) => <option key={'role_' + ROLES[r]} value={ROLES[r]}>{ROLES[r]}</option>)}
+                      </select>
+                      &nbsp;&nbsp;{u.isInstructorAnyCourse() && u.hasRole('student') ? '(TA)' : ''}
+                    </td>
+                  </tr>)
+                })
+              }
+            </tbody>
+          </table>
+          { this.state.profileViewModal
+            ? <ProfileViewModal
+              user={this.state.userToView}
+              done={this.toggleProfileViewModal} />
+          : '' }
+        </div>
       </div>)
```

Comment:
Didn't actually review the React stuff, let's plan for a demo to see what it looks like...